### DISCUSSION TITLE
CK Editor Version and implementing Easy Image.

### DISCUSSION THREAD

**Discussion by Tigerlady** [MasaCMS/MasaCMS#16] [Q&A]
I just updated a development site from Mura to Masa, over all pretty easy.  I do have a few questions.  
What version of CKEditor is this using.  V4 or V5.?  
If this is V4 do you have plans on updating to V5?
Can CKEditors Easy Image be implemented?  (I like this because it can create responsive images.  
V5 also give you the ability to have multiple sources in the code for images.  Responsive is very important these days. 

Because the file manager is no longer available are all images going to just be put into one folder? 
I have also noticed that the url for images is more of a query string rather than the actual path, will this change in the future?

Thank you
Barbara

**Comment by JonasEriksson** [comment]
Barbara, see the first post in this discussion, you need to implement your own license of CKFinder etc. if you want to use it. [https://github.com/MasaCMS/MasaCMS/discussions/6](https://github.com/MasaCMS/MasaCMS/discussions/6) Edit: Also see past (closed) issues: [https://github.com/MasaCMS/MasaCMS/issues/3](https://github.com/MasaCMS/MasaCMS/issues/3) - which is probably where suggestions like "updating jQuery" etc. should go as well, and why not contribute and file it as a Pull Request :-)

**Reply by Tigerlady** [reply]
Jonas,

Yes I am aware that I need to implement my own license. I was more interested in their response as to version of ckeditor and the url issue of images that are copied into the editor.  I also wanted to know if I got a license for Easy Image how well it would work with the current version of the editor.

**Comment by jochemd** [comment]
You can always check [the source code](https://github.com/MasaCMS/MasaCMS/blob/main/core/vendor/ckeditor/CHANGES.md).

While going through all the code to make sure it is compliant with the various copyrights and licensing terms we noticed quite a few of the third party modules had not been updated in a while. We didn't do anything about that at the time because we wanted to keep the changes in the fork as small as possible, so we only made the changes recommended by the legal people. (Except for #8 which I wrote half a year ago and also submitted to Mura at the time.) But I would imagine that updating everything in the vendor folder to the latest minor/patch release is going to happen in the not so distant future because of our commitment to security updates. So that would bring it to [4.16.2](https://ckeditor.com/ckeditor-4/download/)

Updating to CKEditor 5 and adding Easy Image is probably an entirely different story because from the looks of it that is a solution that would upload assets into some image cloud somewhere. I am sure the editor is really easy to implement, but it would mean that everybody who has strict security requirements where every cloud integration is vetted by a legal department would have to go through massive amounts of red tape.

What specifically would you expect from such an integration? Different sizes and web optimized files? Manual cropping? Automatic cropping? AI to focus on faces while cropping? How does your deployment look, are you already in some cloud or is it traditional (VPS) hosting? Clustered or single-server?

**Comment by guustnieuwenhuis** [comment]
> I have also noticed that the url for images is more of a query string rather than the actual path, will this change in the future?

True, this change is part of the replacement for CKFinder.
